brave/src/test/java/brave/secondary_sampling/BasicUsageTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2019-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.secondary_sampling;
+
+import brave.ScopedSpan;
+import brave.Span;
+import brave.Tracing;
+import brave.handler.FinishedSpanHandler;
+import brave.handler.MutableSpan;
+import brave.http.HttpClientHandler;
+import brave.http.HttpServerHandler;
+import brave.http.HttpTracing;
+import brave.propagation.B3Propagation;
+import brave.propagation.TraceContext;
+import brave.secondary_sampling.SecondarySampling.Extra;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+// Do not bring any dependencies into this test without looking at src/it/pom.xml as this is used
+// to verify we don't depend on internals.
+public class BasicUsageTest {
```

Comment:
this test's only point is to hit the main code paths, so that we know we don't depend on classes that don't exist